spec/specreview_checklist.md

```diff
@@ -0,0 +1,73 @@
+# Spec Review Checklist
```

Comment:
Is this checklist file still in play?  If so, there are several changes to modify the "jakarta ee" references to "microprofile".  If not, then we should just remove it.